### PR TITLE
[4.0] Admin modules: Cannot select or filter by position

### PR DESCRIPTION
**Pull request by infograf768**
https://github.com/joomla/joomla-cms/pull/22263 forgot admin modules

### Summary of Changes
Change type of the position field for admin modules.
Setting the correct client id.

### Testing Instructions
Install j4 from 4.0-dev or nightly.
Display Administrator Modules Manager
Create or edit one these modules.

Note: there are other errors 
### Expected result
The `Position` field should be similar to the one used for Site Modules, i.e. letting clear, select, set to None the position.
Filtering by position should display the administrator template positions

### Actual result
For the manager, positions are sometimes the ones from the frontend template in the filter
screen shot 2018-12-16 at 17 36 51

Editing the admin module displays a simple field
screen shot 2018-12-16 at 17 37 09

### After patch

screen shot 2018-12-16 at 17 26 51

screen shot 2018-12-16 at 17 27 13

### Note
There are other errors, not only related to admin modules.
1. The calendar icons are showing over the position dropdown (see image above)
2. After filtering in the manager by anything else than Select and clicking anywhere on the page, the filter fields are displayed horizontally on the page, hiding some items.
screen shot 2018-12-16 at 17 40 57

### General Comment
The decision to take off from the Modules Manager the Site/Admin filter is at the origin of some issues concerning `userstate` and filters specially when a user switches from one to the other via the menu or System CPanel.

@Fedik

### CODE REVIEW

File: administrator/components/com_modules/forms/moduleadmin.xml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="utf-8"?>
 <form>
-	<fieldset>
+	<fieldset addfieldprefix="Joomla\Component\Modules\Administrator\Field">>
```
Comment (Quy) [inline administrator/components/com_modules/forms/moduleadmin.xml]:
```suggestion
	<fieldset addfieldprefix="Joomla\Component\Modules\Administrator\Field">
```

Comment (infograf768) [reply administrator/components/com_modules/forms/moduleadmin.xml]: done. Thanks

File: administrator/components/com_modules/tmpl/module/edit.php [hunk 2]
```diff
@@ -43,6 +43,12 @@
 
 $input = Factory::getApplication()->input;
 
+// In case of admin module
+if ($this->item->client_id == 1)
+{
+	$input->set('client_id', 1);
```
Comment (laoneo) [inline administrator/components/com_modules/tmpl/module/edit.php]: Input shouldn't be modified in a layout file.

Comment (infograf768) [reply administrator/components/com_modules/tmpl/module/edit.php]: @laoneo 
What do you suggest to solve the issue?

Comment (infograf768) [reply administrator/components/com_modules/tmpl/module/edit.php]: The other solution is to create specific fields for admin modules.
I could not find another way.

Comment (infograf768) [reply administrator/components/com_modules/tmpl/module/edit.php]: Another solution: doing that in the models.
On it now.

Comment (infograf768) [reply administrator/components/com_modules/tmpl/module/edit.php]: Done. Please check.

**Comment by SharkyKZ** [comment]
Is this not the same as #22040? At least the filter part.

**Comment by infograf768** [comment]
not really as we have a new field and the modulesposition field was modified in 4.0.
see 
https://github.com/joomla/joomla-cms/pull/22263

File: administrator/components/com_modules/tmpl/module/edit.php (hunk 2 above)
Comment (Fedik) [reply administrator/components/com_modules/tmpl/module/edit.php]: I think a correct way would be to use a field attribute.
but for this also need to improve `ModulesPositioneditField` also

**Comment by Fedik** [comment]
> The calendar icons are showing over the position dropdown

that is CSS issue, somewhere, I tried to fix it in https://github.com/joomla/joomla-cms/pull/22263/files#diff-712f8768cba688c1b024f23f9941f51eR4 not sure why it there again

**Comment by infograf768** [comment]
@SharkyKZ 
You are right. Once again one of these aspects of 4.0 which is way behind staging.
What shall we do? 
What is the best code?
If it is the one you used for 3.9.0, I can modify this patch, but I also would have to apply the changes from your PR here for all to work OK.

@wilsonge

File: administrator/components/com_modules/tmpl/module/edit.php (hunk 2 above)
Comment (infograf768) [reply administrator/components/com_modules/tmpl/module/edit.php]: see below

**Comment by infograf768** [comment]
@Fedik 
If we use the code in 3.9.0 from @SharkyKZ then indeed we just have to modify the way to get the `$clientId` in `ModulesPositioneditField.php` by using `getUserState('com_modules.modules.client_id', 0, 'int')`

Concerning the dropdown z-index, it looks like `.choices__list--dropdown` in template.css is loaded BEFORE the `/media/vendor/choicesjs/css/choices.css` and is therefore overriden by that file. (Debug on).

So we get
```
.choices__list--dropdown {
  display: none;
  z-index: 1;
  position: absolute;
  width: 100%;
  background-color: #FFFFFF;
  border: 1px solid #DDDDDD;
  top: 100%;
  margin-top: -1px;
  border-bottom-left-radius: 2.5px;
  border-bottom-right-radius: 2.5px;
  overflow: hidden;
  word-break: break-all;
}
```
and `z-index = 1;` instead of 10

**Comment by SharkyKZ** [comment]
That should work. Just don't add 3rd argument to `getUserState()`. I added it by mistake. Oops.

**Comment by infograf768** [comment]
Modified to use `getUserState`
Please test.

**Comment by ReLater** [comment]
I have tested this item :white_check_mark: successfully on c4d5e9ea98f24e6f01659350c9ddd71d00bf97d9<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23289">issues.joomla.org/tracker/joomla-cms/23289</a>.</sub>

**Comment by anibalsanchez** [comment]
I have tested this item :white_check_mark: successfully on 76b957b7af1fef54e371ca7e395233e3ab60d433<br /><br />Test OK<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23289">issues.joomla.org/tracker/joomla-cms/23289</a>.</sub>

**Comment by infograf768** [comment]
RTC. Thanks for testing.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23289">issues.joomla.org/tracker/joomla-cms/23289</a>.</sub>

**Comment by wilsonge** [comment]
Thanks guys!

**Pull request merged**